compiler/noirc_frontend/src/node_interner.rs

```diff
@@ -1370,8 +1370,14 @@ impl NodeInterner {
         type_bindings: &mut TypeBindings,
         recursion_limit: u32,
     ) -> Result<TraitImplKind, Vec<TraitConstraint>> {
-        let make_constraint =
-            || TraitConstraint::new(object_type.clone(), trait_id, trait_generics.to_vec());
+        let make_constraint = || {
+            TraitConstraint::new(
+                object_type.clone(),
+                trait_id,
+                trait_generics.to_vec(),
+                Span::default(),
```

Comment:
Can this `Span` be non-empty for items in `self.trait_implementation_map`?

Reply:
`trait_implementation_map` doesn't hold any constraints? This is only used for returning the constraint whose impl could not be found. The actual span of the constraint is supplied in `verify_trait_constraint` when building the `NoMatchingImplFoundError`.